pdu/UDH.go

```diff
@@ -90,15 +84,21 @@ func (u UDH) FindInfoElement(id byte) (ie *InfoElement, found bool) {
 
 // GetConcatInfo return concatenated message info, return 0 if
 // Concat Message InfoElement is not found in the UDH
-func (u UDH) GetConcatInfo() (totalParts, partNum, mref int, found bool) {
-	if len(u) == 0 {
-		found = false
-		return
-	}
-	if ie, found := u.FindInfoElement(data.UDH_CONCAT_MSG_8_BIT_REF); found && len(ie.Data) == 3 {
-		mref = int(ie.Data[0])
-		totalParts = int(ie.Data[1])
-		partNum = int(ie.Data[2])
+func (u UDH) GetConcatInfo() (totalParts, sequence byte, reference uint16, found bool) {
+	for _, element := range u {
+		options := element.Data
+		switch element.ID {
+		case data.UDH_CONCAT_MSG_8_BIT_REF:
+			reference = uint16(options[0])
```

Comment:
Please check options length here. We dont want panic happen because of out of range